src/transformers/modeling_utils.py

```diff
@@ -3922,15 +3941,12 @@ def _find_mismatched_keys(
                     ignore_mismatched_sizes,
                 )
                 if low_cpu_mem_usage:
-                    if is_fsdp_enabled() and not is_local_dist_rank_0():
+                    if is_fsdp_enabled() and not is_local_dist_rank_0() and not is_quantized:
                         for key, param in model_to_load.state_dict().items():
                             if param.device == torch.device("meta"):
-                                if hf_quantizer is None:
-                                    set_module_tensor_to_device(
-                                        model_to_load, key, "cpu", torch.empty(*param.size(), dtype=dtype)
-                                    )
-                                else:
-                                    hf_quantizer.create_quantized_param(model, param, key, "cpu", state_dict)
```

Comment:
Just to make sure I've understood - we don't need this anymore as we move creating quantized params when loading in the state dict?

Reply:
`hf_quantizer is None`, i.e., not quantized will always be False as the conditional logic `if is_fsdp_enabled() and not is_local_dist_rank_0() and not is_quantized` has this and as such this inner conditional is no longer required. The quantized parameters are initilaized in the else logic on line 3950